src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -276,23 +339,31 @@ public Cci.IUnitReference GetUnit(EmitContext context)
 
         public string NamespaceName => string.Empty;
 
-        internal static string GenerateDataFieldName(ImmutableArray<byte> data)
+        private static string HashToHex(ImmutableArray<byte> data)
         {
-            var hash = CryptographicHashProvider.ComputeSourceHash(data);
+            ImmutableArray<byte> hash = CryptographicHashProvider.ComputeSourceHash(data);
+
+#if NETCOREAPP2_1_OR_GREATER
+            return string.Create(hash.Length * 2, hash, (destination, hash) => toHex(hash, destination));
+#else
             char[] c = new char[hash.Length * 2];
-            int i = 0;
-            foreach (var b in hash)
+            toHex(hash, c);
+            return new string(c);
+#endif
+
+            static void toHex(ImmutableArray<byte> source, Span<char> destination)
             {
-                c[i++] = Hexchar(b >> 4);
-                c[i++] = Hexchar(b & 0xF);
+                int i = 0;
+                foreach (var b in source.AsSpan())
```

Comment:
>source.AsSpan() [](http://example.com/codeflow?start=34&length=15)

Is there a specific reason behind calling ```AsSpan``` here?

Reply:
There's less overhead to enumerating a span, as it's recognized by the runtime, bounds checks are eliminated, etc. e.g.
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System.Collections.Immutable;
using System.Linq;

[MemoryDiagnoser]
public partial class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    private ImmutableArray<byte> _bytes = ImmutableArray.CreateRange(Enumerable.Range(0, 100).Select(i => (byte)i));

    [Benchmark(Baseline = true)]
    public int Sum()
    {
        int sum = 0;
        foreach (byte b in _bytes) sum += b;
        return sum;
    }

    [Benchmark]
    public int SumSpan()
    {
        int sum = 0;
        foreach (byte b in _bytes.AsSpan()) sum += b;
        return sum;
    }
}
```
on my machine:

|  Method |     Mean |
|-------- |---------:|
|     Sum | 52.85 ns |
| SumSpan | 37.12 ns |